Title       : x11-base/xorg-server-1.3.0.0 Xephyr segfaults on startup on amd64
Assignee    : x11@gentoo.org
Reported    : 2007-09-20 17:22 0000
Updated     : 2007-09-20 17:52:21 0000
Status      : NEW
URL         : https://bugs.freedesktop.org/show_bug.cgi?id=11582
Severity    : normal
Priority    : P2
Reporter    : chithanh@cs.tu-berlin.de
Product     : Gentoo Linux
Component   : Server
Keywords    : Inclusion
Comments    : 2
Attachments : 1
URL         : http://bugs.gentoo.org/show_bug.cgi?id=193207

commit dcc3de91d2b80be98e4488df29ec6b551c7ff6d1
Author: Alan Coopersmith <alan.coopersmith@sun.com>
Date:   Sat Aug 18 11:50:33 2007 +0200

    Xephyr: fix immediat segfault on amd64

    This closes bug https://bugs.freedesktop.org/show_bug.cgi?id=11582

diff --git a/hw/kdrive/ephyr/hostx.c b/hw/kdrive/ephyr/hostx.c
index d0a2f2f..12118da 100644
--- a/hw/kdrive/ephyr/hostx.c
+++ b/hw/kdrive/ephyr/hostx.c
@@ -40,6 +40,7 @@
 #include <X11/Xatom.h>
 #include <X11/keysym.h>
 #include <X11/extensions/XShm.h>
+#include <X11/Xmd.h>
 
 /*  
  * All xlib calls go here, which gets built as its own .a .
@@ -79,7 +80,7 @@ static int            HostXWantDamageDebug = 0;
 
 extern KeySym         EphyrKeymap[];
 
-extern KeySym	      kdKeymap[];
+extern CARD32	      kdKeymap[];
 extern int	      kdMinScanCode;
 extern int	      kdMaxScanCode;
 extern int	      kdMinKeyCode;
